src/ui/studio/save-creation/index.tsx

```diff
@@ -415,12 +431,49 @@ const UploadForm = ({ uploadState, handleUpload }) => {
     }
   }
 
+  async function handleSelectChange(newValue: SingleValue<SeriesOption>) {
+    let newSeriesValue = newValue ? newValue.value : null;
+    dispatch({ type: 'UPDATE_SERIES', value: newSeriesValue });
+  }
+
+  const filterSeries = async(inputValue: string) => {
+    const seriesList: SeriesOption[] = [];
+    const ocSeries = await opencast.getSeries(inputValue);
+    for (const [key, value] of ocSeries ) {
+      seriesList.push({ value: key, label: value });
+      if (seriesId && seriesId === key) {
+        seriesDefault = { value: key, label: value };
+      }
+    }
+    return seriesList;
+  };
+
+  const lastTimeout = useRef<ReturnType<typeof setTimeout> | null>(null);
+
+  const loadSeriesOptions = (inputValue: string) => {
+    const seriesOptions = new Promise<SeriesOption[]>(resolve => {
+      if (lastTimeout.current !== null) {
+        clearTimeout(lastTimeout.current);
+      }
+      lastTimeout.current = setTimeout(() => {
+        resolve(filterSeries(inputValue));
+      }, 30);
+    });
```

Comment:
I would either extract the debounce logic into a separate function or just use `lodash.debounce`. That package is already in our dependency tree, so we might as well just add it to `package.json` and use it ourselves.